### DISCUSSION TITLE
Future works

### DISCUSSION THREAD

**Discussion by JakNowy** [igorbenav/fastcrud#103] [General]
From my experience using FastCRUD in my project, the improvements I would love to see the most are:

- streamlined router paths
- ability to override router methods (for example implement get_joined() in router GET)
- have multiple levels of nesting in joins

Do you have any sort of roadmap for future months? I might be able to work on one of this maybe.

Also I honestly think this library is good enough to advertise it somewhere like Linkedin, where we might even tag Sebastian Ramirez to extend the audience. This would hopefully increase the number of users and maybe even contributors if we state that features are still being improved and help is welcome.

**Comment by JakNowy** [comment]
CC @igorbenav

**Reply by igorbenav** [reply]
I'll elaborate better later today. Sorry for the delay btw, I was traveling and couldn't answer properly

**Comment by igorbenav** [comment]
> Also I honestly think this library is good enough to advertise it somewhere like Linkedin, where we might even tag Sebastian Ramirez to extend the audience. This would hopefully increase the number of users and maybe even contributors if we state that features are still being improved and help is welcome.

I did it a few times actually, [here](https://www.linkedin.com/posts/igor-magalhaes-r_github-igorbenavfastcrud-fastcrud-is-activity-7171867300201906176-QZw4?utm_source=share&utm_medium=member_desktop), [here](https://www.linkedin.com/posts/igor-magalhaes-r_github-igorbenavfastcrud-fastcrud-is-activity-7158589728139755520-CH3I?utm_source=share&utm_medium=member_desktop), [here](https://www.linkedin.com/posts/igor-magalhaes-r_github-igorbenavfastcrud-fastcrud-is-activity-7156616797025153024-5V55?utm_source=share&utm_medium=member_desktop) (linkedin), on reddit as well [here](https://www.reddit.com/r/Python/comments/1abwqni/fastcrud_powerful_crud_methods_and_automatic/), [here](https://www.reddit.com/r/FastAPI/comments/1ab5pj5/fastcrud_powerful_crud_methods_and_automatic/) and [here](https://www.reddit.com/r/Backend/comments/1abxeke/fastcrud_powerful_crud_methods_and_automatic/). Plus, a tutorial on [medium](https://medium.com/@igorbenav/prototyping-fastapi-faster-with-fastcrud-5676a0499e97) and [dev.to](https://dev.to/igorbenav/prototyping-fastapi-faster-with-fastcrud-4ko8). Sharing it on linkedin and reddit is great every couple of months, but I think the actual best way to get people really using it is creating tutorials (not necessarily fastcrud tutorials, but just regular fastapi tutorials that use fastcrud).

It's actually crazy to think it's been just 5 months since launch, I'll create more linkedin / reddit posts, I'll see if I can write more tutorials as well.

**Comment by igorbenav** [comment]
> From my experience using FastCRUD in my project, the improvements I would love to see the most are:
> 
> * streamlined router paths
> * ability to override router methods (for example implement get_joined() in router GET)
> * have multiple levels of nesting in joins

I think the most relevant ones for now are indeed (1) creating an elegant way to add functionality or completely change endpoints as we discussed [here](https://github.com/igorbenav/fastcrud/pull/80) and [here](https://github.com/igorbenav/fastcrud/discussions/109#discussioncomment-9845223) and (2) the streamlined endpoints. I'm not planning too much ahead since new issues arise and the urgency changes, but multiple nesting for joins would come next with stuff like #50.

I feel like I'll need to rewrite the nested joins part or at least make it simpler to work better with multiple levels of nesting.
